rotkehlchen/ethchain.py

```diff
@@ -53,17 +53,43 @@ def attempt_connect(self, ethrpc_port, mainnet_check=True) -> Tuple[bool, str]:
                         'the ethereum mainnet'.format(ethrpc_port)
                     )
                     return False, message
+                if self.web3.eth.syncing:  # pylint: disable=no-member
+                    curr_block = self.web3.eth.syncing.currentBlock  # pylint: disable=no-member
+                    high_block = self.web3.eth.syncing.highestBlock  # pylint: disable=no-member
+                    return self.is_syncronized(curr_block, high_block)
+                else:
+                    high_block = self.get_eth_highest_block()
+                    curr_block = self.web3.eth.blockNumber  # pylint: disable=no-member
+                    return self.is_syncronized(curr_block, high_block)
 
             self.connected = True
             return True, ''
         else:
-            logger.warn('Could not connect to a local ethereum node. Will use etherscan only')
+            logger.warning('Could not connect to a local ethereum node. Will use etherscan only')
             self.connected = False
             message = 'Failed to connect to ethereum node at port {}'.format(ethrpc_port)
 
         # If we get here we did not connnect
         return False, message
 
+    def is_syncronized(self, curr_block, high_block):
+        """ Validate that the local ethereum node is syncronized
+            at least 99.99% before using local blockchain
+
+        Returns a tuple (results, message)
+            - result: Boolean for confirmation of syncronized
+            - message: A message containing information on what the status is. """
+        sync_perc = (100 * curr_block / high_block)
+        if sync_perc < 99.99:
```

Comment:
Playing with floating numbers is not required here. Just set a specific block number tolerance. Let's say if it's within 20 blocks of the high block it should be fine

Reply:
Great, will do!